### ISSUE TITLE
--use-attention-mask flag negatively impacts perplexity results

### ISSUE THREAD

**Issue by Alex-Vasile** [nod-ai/shark-ai#1495]
Compiling an unsharded model with and without the `--use-attention-mask` flag and then running perplexity on it results in different values. Below are results for LLama 8B.
```
Llama3 8b, bs=4, IREE ppl

unsharded 
with attn mask - 13.356212
{
  "perplexities": [
    7.410479,
    22.796532,
    10.047695,
    13.17014
  ],
  "mean_perplexity": 13.356212
}
w/o attn mask - 13.016442
{
  "perplexities": [
    7.182783,
    21.818655,
    9.878222,
    13.18611
  ],
  "mean_perplexity": 13.016442
}
```
@archana-ramalingam 

A difference was also found with a toy deepseek model 2263 vs 2266 (model weights are random so actual values are meaningless). For deepseek only the prefill logits were affected, the decode logits were identical.

**Comment by Alex-Vasile** [comment]
Update: Does not appear cause a difference in eager mode. This is a tracing/IREE issue. Working on getting a smaller repro.

**Comment by Alex-Vasile** [comment]
Actual issue was that the perplexity test was grabbing the wrong prefill logits, specifically it was looking at the padded logits, whose values are undefined and so would likely change since the two approach (with and without the flag) use a slightly different mask.

**Comment by archana-ramalingam** [comment]
We are seeing this on eager perplexity test too. 
Llama8b fp16, decode logits only match: 0.093%, while prefill matches 100% and perplexity score deviates largely. (Will reproduce eager numbers once regression is fixed ToM.)
